react/lesson1.md

```diff
@@ -16,17 +16,72 @@
 
 # React
 
-[React](https://facebook.github.io/react/) is a library by Facebook.
+[React](https://facebook.github.io/react/) is a library by Facebook for building user interfaces. 
+
+Today, the focus is on component based architectures, and React provides a thin layer on top of JavaScript and HTML to allow you to build custom components that can encapsulate behavior and style and can be re-used throughout your application and even exported for others to use.
+
+We already have some existing components in HTML. When you write this:
+
+```html
+<input type="number" name="age" value="5">
+```
+
+..it generates this: 
+
+<img src="assets/html_input.png" width="150">
+
+An "input" box, that only supports numbers, with a set of up/down arrows for changing the value, and a pre-defined value of 5. So the **element tag** tells the browser which element to render, and the attributes specify the settings for that element and any information we want to pass into it.
 
 ## Why React?
-ToDo
+
+Besides providing the tools to build powerful and well encapsulated components, React also takes care of updating the view to match our data.
+
+Suppose you make an HTTP GET request to "http://blog.com/post?id=123" and you store the result in a variable:
+
+```javascript
+var post = {id: '123', title: 'Post 1', content: 'Lorem ipsum 1'};
+```
+
+And our HTML looks like this:
+
+```html
+<h1 id="title"> </h1>
+<p id="content"> </p>
+```
+
+As we've learned, we can dinamically insert content into our HTML using JavaScript:
```

Comment:
dynamically

Reply:
I'll just run the whole thing through an type-checker 😛